romancal/multiband_catalog/detection_image.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import warnings
+
+import numpy as np
+from astropy.convolution import convolve_fft
+from astropy.utils.exceptions import AstropyUserWarning
+from photutils.segmentation import make_2dgaussian_kernel
+
+from romancal.datamodels import ModelLibrary
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def make_gaussian_kernel(fwhm):
+    """
+    Make a normalized 2D circular Gaussian kernel.
+
+    The kernel will have odd sizes in both X and Y, be centered in the
+    central pixel, and normalized to sum to 1.
+
+    Parameters
+    ----------
+    fwhm : float
+        The full-width at half-maximum (FWHM) in pixels of the 2D
+        Gaussian kernel.
+
+    Returns
+    -------
+    kernel : `astropy.convolution.Kernel2D`
+        The output Gaussian kernel.
+    """
+    size = np.ceil(3 * fwhm).astype(int)
+    size = size + 1 if size % 2 == 0 else size  # make size be odd
+    kernel = make_2dgaussian_kernel(fwhm, size=size)  # sums to 1
+    return kernel.array
+
+
+def make_det_image(library, kernel_fwhm):
+    """
+    Make a detection image from a library of models.
+
+    The detection image is the weighted sum of the input models, where
+    the weights are SED weight divided by the read noise variance.
+
+    Parameters
+    ----------
+    library : `romancal.datamodels.ModelLibrary`
+        The input library of models.
+
+    kernel_fwhm : float
+        The full-width at half-maximum (FWHM) in pixels of the 2D
+        Gaussian kernel used to smooth the detection image.
```

Comment:
And maybe not for right now, but remembering my thinking... for a PSF, we want to use the appropriate PSF for each band, so a different FWHM for each band.  For an extended object, it's less important, but technically we would want a true source profile convolved with the appropriate PSF.